docs/apache-airflow/concepts/operators.rst

```diff
@@ -127,12 +132,12 @@ Deep nested fields can also be substituted, as long as all intermediate fields a
         # [additional code here...]
 
 
-    t = PythonOperator(
-        task_id="transform_data",
-        python_callable=transform_data,
-        op_args=[MyDataTransformer(MyDataReader("/tmp/{{ ds }}/my_file"))],
-        dag=dag,
-    )
+    @task(task_id="transform_data")
+    def transform_data(data_transformer):
+        pass
+
+
+    t = transform_data([MyDataTransformer(MyDataReader("/tmp/{{ ds }}/my_file"))])
```

Comment:
I don’t think this works, XComArg inputs are not automatically processed in taskflow.

This needs to be implemented as

```python
@task(task_id="transform_data")
def transform_data(ds=None):
    MyDataTransformer(MyDataReader("/tmp/{ds}/my_file"))
```

but this would make the section above meaningless. So a better way would be to use another operator altogether, not `PythonOperator`, for the example.

Reply:
Why you think `PythonOperator` wouldn't work in this case @uranusjr ? What do you think would be the better operator here? 
I am new to this Operator part.

Reply:
BashOperator is probably a reasonable choice. PythonOperator works as well, but `@task` does not (it does not render templates; only operator _classes_ do this).

Maybe we can revert this change back to PythonOperator in this PR, and rewrite it later.

Reply:
I see @uranusjr 
Then saying `@task` decorator as alternative to classic usage of Operator is not correct in this case. Am i right?

Reply:
Yea that's a good idea. I believe it is mentioned elsewhere, but it does not hurt to mention it here again.